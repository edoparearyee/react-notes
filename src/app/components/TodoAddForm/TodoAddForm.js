import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './TodoAddForm.scss';

class TodoAddForm extends Component {
  render() {
    return (
      <form className="TodoAddForm" onSubmit={e => this.handleSubmit(e)}>
        <input
          className="TodoAddForm__Title"
          ref={c => (this._inputTitle = c)}
          onInput={e => this.handleTitleInput(e)}
          onChange={() => null}
          name="title"
          type="text"
          value={this.props.value}
        />
        <button type="submit" disabled={!this.props.value.length}>
          Add
        </button>
      </form>
    );
  }

  handleTitleInput(event) {
    this.props.onInput({
      value: event.target.value,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.onAdd({
      title: this.props.value,
    });
  }
}

TodoAddForm.propTypes = {
  value: PropTypes.string.isRequired,
  onAdd: PropTypes.func,
  onInput: PropTypes.func,
};

export default TodoAddForm;
