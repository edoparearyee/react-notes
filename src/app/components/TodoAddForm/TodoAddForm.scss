.TodoAddForm {
  align-items: flex-end;
  display: Flex;
  justify-content: flex-start;
  width: 100%;
}

.TodoAddForm__Title {
  width: 100%;
}

.TodoAddForm__Input {
  flex-grow: 2;
  margin-right: 10px;
}
