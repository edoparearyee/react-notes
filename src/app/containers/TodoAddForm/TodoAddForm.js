import { connect } from 'react-redux';
import { todoAdd } from '../../actions/todos';
import { formEdit, formClear } from '../../actions/form';

import TodoAddForm from '../../components/TodoAddForm/TodoAddForm';

const mapStateToProps = state => {
  return {
    value: state.form.value,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onAdd: ({ title }) => {
      dispatch(todoAdd(title));
      dispatch(formClear());
    },
    onInput: ({ value }) => {
      dispatch(formEdit(value));
    },
  };
};

const TodoAddFormContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(TodoAddForm);

export default TodoAddFormContainer;
