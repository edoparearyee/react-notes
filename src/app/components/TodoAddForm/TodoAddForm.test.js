import React from 'react';
import ReactDOM from 'react-dom';

import TodoAddForm from './TodoAddForm';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TodoAddForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});
