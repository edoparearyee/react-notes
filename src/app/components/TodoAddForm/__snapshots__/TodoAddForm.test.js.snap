// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoAddForm renders without crashing 1`] = `
<form
  className="TodoAddForm"
  onSubmit={[Function]}
>
  <input
    className="TodoAddForm__Title"
    name="title"
    onChange={[Function]}
    onInput={[Function]}
    type="text"
    value="foo"
  />
  <button
    disabled={false}
    type="submit"
  >
    Add
  </button>
</form>
`;
